ls1intum/Athena
# Update protobuf version for GitHub Actions build

I had a problem with the prebuilt images from the GitHub package registry:

`docker run ghcr.io/ls1intum/athene/athene-load-balancer:latest`

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
[...]
  File "/usr/src/app/./src/main.py", line 14, in <module>
    import src.clustering_pb2 as Protobuf
  File "/usr/src/app/./src/clustering_pb2.py", line 36, in <module>
    _descriptor.FieldDescriptor(
  File "/usr/local/lib/python3.10/site-packages/google/protobuf/descriptor.py", line 560, in __new__
    _message.Message._CheckCalledFromGeneratedFile()
TypeError: Descriptors cannot not be created directly.
If this call came from a _pb2.py file, your generated code is out of date and must be regenerated with protoc >= 3.19.0.
If you cannot immediately regenerate your protos, some other possible workarounds are:
 1. Downgrade the protobuf package to 3.20.x or lower.
 2. Set PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python (but this will use pure-Python parsing and will be much slower).

More information: https://developers.google.com/protocol-buffers/docs/news/2022-05-06#python-updates
```

This seems to be related to an upgrade of the `protobuf` package.

This PR updates the `protobuf` version in the GitHub Actions workflow to mitigate the problem.

# Testing
To test this fix, first run:
```docker run ghcr.io/ls1intum/athene/athene-load-balancer:latest```

You should see the error from above.

Then, run:
```docker run ghcr.io/ls1intum/athene/athene-load-balancer:protobuf-build-version```

This is the image built from this branch. The error should be gone now.
